CONTRIBUTING.md

```diff
@@ -3,104 +3,190 @@
+      <br/> `Introduce nvim namespace: Contrib YCM: Fix build dir calculation`
         - Subtasks can be *activity-oriented* (doing different things on the same area)
           or *scope-oriented* (doing the same thing on different areas).
     - Granularity helps, but it's conceptual size that matters, not extent size.
-- Use the imperative voice: "Fix bug" rather than "Fixed bug" or "Fixes bug."
+- Use the [imperative voice][imperative]: "Fix bug" rather than "Fixed bug" or "Fixes bug."
+
+### Reviewing pull requests
+
+Using a checklist during reviews is highly recommended, so we [provide one at
+the wiki][wiki-review-checklist]. If you think it could be improved, feel free
+to edit it.
+
+Reviewing can be done on GitHub, but you may find it easier to do locally.
+Using [`hub`][hub], you can do the following to create a new branch with the
+contents of a pull request, such as [#1820][github-pr-1820]:
 
+    hub checkout https://github.com/neovim/neovim/pull/1820
+
+Use [`git log -p master..FETCH_HEAD`][git-history-filtering] to list all
+commits in the feature branch which aren't in the `master` branch; `-p`
+shows each commit's diff. To show the whole surrounding function of a change
+as context, use the `-W` argument as well.
+
+You may find it easier to instead use an interactive program for code reviews,
+such as [`tig`][tig].
 
 [cla]: https://docs.google.com/forms/d/1u54bpbwzneDIRltFx1TGi2evKxY3w0cOV3vlpj8DPbg/viewform
-[clint]: clint.py
 [clang-format]: http://clang.llvm.org/docs/ClangFormat.html
-[entry]: https://github.com/neovim/neovim/issues?labels=entry-level&state=open
+[clint]: clint.py
+[entry-level]: https://github.com/neovim/neovim/issues?labels=entry-level&state=open
+[gcc-warnings]: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
+[git-bisect]: http://git-scm.com/book/tr/v2/Git-Tools-Debugging-with-Git
+[git-feature-branch]: https://www.atlassian.com/git/tutorials/comparing-workflows
+[git-history-filtering]: https://www.atlassian.com/git/tutorials/git-log/filtering-the-commit-history
+[git-history-rewriting]: http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
+[git-rebasing]: http://git-scm.com/book/en/v2/Git-Branching-Rebasing
+[github-issues]: https://github.com/neovim/neovim/issues
+[github-pr-1820]: https://github.com/neovim/neovim/pull/1820
+[hub]: https://hub.github.com/
 [imperative]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
-[style]: http://neovim.org/develop/style-guide.xml
+[style-guide]: http://neovim.org/develop/style-guide.xml
+[tig]: https://github.com/jonas/tig
+[valgrind]: http://valgrind.org/
+[vcs]: https://en.wikipedia.org/wiki/Revision_control
 [waffle]: https://waffle.io/neovim/neovim
-[wiki]: https://github.com/neovim/neovim/wiki/Contributing
+[wiki-building-testing]: https://github.com/neovim/neovim/wiki/Building-Neovim#testing
+[wiki-building-troubleshooting]: https://github.com/neovim/neovim/wiki/Building-Neovim#troubleshootingfaq
```

Comment:
Should be `https://github.com/neovim/neovim/wiki/Building-Neovim#troubleshooting`